service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,164 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.converter.GiftCertificateDTOToEntityConverter;
+import com.epam.esm.converter.GiftCertificateEntityToDTOConverter;
+import com.epam.esm.dto.GiftCertificateDTO;
+import com.epam.esm.exception.ServiceException;
+import com.epam.esm.hibernate.CertificateRepository;
+import com.epam.esm.persistence.GiftCertificateEntity;
+import com.epam.esm.persistence.TagEntity;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.validator.PreparedValidatorChain;
+import com.epam.esm.validator.realisation.IntermediateCertificateLink;
+import com.epam.esm.validator.realisation.IntermediateSortLink;
+import com.epam.esm.validator.realisation.certificate.*;
+import com.epam.esm.validator.realisation.sort.FieldValidatorLink;
+import com.epam.esm.validator.realisation.sort.MethodValidatorLink;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class GiftCertificateServiceImpl implements GiftCertificateService {
+
+    private final GiftCertificateEntityToDTOConverter converterToDto;
+    private final GiftCertificateDTOToEntityConverter converterToEntity;
+    private final CertificateRepository repository;
+
+    @Autowired
+    public GiftCertificateServiceImpl(GiftCertificateEntityToDTOConverter converterToDto,
+                                      GiftCertificateDTOToEntityConverter converterToEntity,
+                                      CertificateRepository repository) {
+        this.converterToDto = converterToDto;
+        this.converterToEntity = converterToEntity;
+        this.repository = repository;
+    }
+
+    @Override
+    public List<GiftCertificateDTO> findAll(int limit, int page) {
+        return createResultList(repository.findAll(limit, page));
+    }
+
+    @Override
+    public Optional<GiftCertificateDTO> find(long id) {
+        Optional<GiftCertificateEntity> certificateFromDao = repository.find(id);
+        return certificateFromDao.map(converterToDto);
+    }
+
+    @Override
+    public int create(GiftCertificateDTO certificateDTO) {
+        PreparedValidatorChain<GiftCertificateDTO> chain = new IntermediateCertificateLink();
+        chain.linkWith(new CertificateNameValidatorLink())
+                .linkWith(new DescriptionValidatorLink())
+                .linkWith(new DurationValidatorLink())
+                .linkWith(new PriceValidatorLink())
+                .linkWith(new TagsValidatorLink());
+        if (chain.validate(certificateDTO)) {
+            GiftCertificateEntity certificate = converterToEntity.apply(certificateDTO);
+            return repository.create(certificate);
+        }
+        return 0;
```

Comment:
Returning 0 seems weird. How about replace it with Exception throwing? I suppose this is absolutely incorrect app behavior and an error should be thrown